src/content/posts/2021/pitfalls-of-unoptimzied-nuget-packages/index.md

```diff
@@ -0,0 +1,177 @@
+
+* Also, if you have a local which is never used at all, then there is no storage allocated for it if the flag is set.
+* Similarly, the compiler is more aggressive about re-using the unnamed temporary slots sometimes used to store results of subexpression calculations.
+* Also, with the flag set the compiler is more aggressive about generating code that throws away “temporary” values quickly for things like controlling variables of switch statements, the condition in an “if” statement, the value being returned, and so on. In the non-optimized build these values are treated as unnamed local variables, loaded from and stored to specific locations. In the optimized build they can often be just kept on the stack proper.
+* We eliminate pretty much all of the “breakpoint convenience” no-ops.
+* If a try block is empty then clearly the catch blocks are not reachable and can be trimmed. (Finally blocks of empty tries are preserved as protected regions because they have unusual behaviour when faced with certain exceptions; see the comments for details.) 
+* If we have an instruction which branches to `LABEL1`, and the instruction at `LABEL1` branches to `LABEL2`, then we rewrite the first instruction as a branch straight to `LABEL2`. Same with branches that go to returns.
+* We look for “branch over a branch” situations. For example, here we go to `LABEL1` if condition is false, otherwise we go to `LABEL2`.
+
+```
+brfalse condition, LABEL1
+br LABEL2
+LABEL1: somecode
+```
+
+Since we are simply branching over another branch, we can rewrite this as simply "if condition is true, go to `LABEL2`":
+
+```
+brtrue condition, LABEL2
+somecode
+```
+
+* We look for “branch to nop” situations. If a branch goes to a nop then you can make it branch to the instruction after the nop.
+* We look for “branch to next” situations; if a branch goes to the next instruction then you can eliminate it.
+* We look for two return instructions in a row; this happens sometimes and obviously we can turn it into a single return instruction.
+
+Since writing this, Eric Lippert left Microsoft and the C# compiler was rewritten and open sourced.
+You can browse the [Roslyn's](http://github.com/dotnet/roslyn) repository and you'll [discover that new optimizations are still being suggested](https://github.com/dotnet/roslyn/issues?q=is%3Aissue+is%3Aopen+optimization+proposal) and added.
+
+## Sensible defaults
+
+> Optimize also tells the common language runtime to optimize code at runtime. By default, optimizations are disabled. Specify Optimize+ to enable optimizations. When building a module to be used by an assembly, use the same Optimize settings as used by the assembly. It's possible to combine the Optimize and Debug options.
```

Comment:
Where do I specify `Optimze+`? It doesn't seem like an actionable advice for me if I'm using msbuild and not csc directly.